control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3494,3 +3502,239 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
+			Type:    string(hyperv1.AWSDefaultSecurityGroupCreated),
+			Status:  metav1.ConditionTrue,
+			Message: hyperv1.AllIsWellMessage,
+			Reason:  hyperv1.AsExpectedReason,
+		}
+		hcp.Status.Platform = &hyperv1.PlatformStatus{
+			AWS: &hyperv1.AWSPlatformStatus{
+				DefaultWorkerSecurityGroupID: sgID,
+			},
+		}
+	}
+	meta.SetStatusCondition(&hcp.Status.Conditions, *condition)
+
+	if err := r.Client.Status().Patch(ctx, hcp, client.MergeFromWithOptions(originalHCP, client.MergeFromWithOptimisticLock{})); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return creationErr
+}
+
+func awsSecurityGroupFilters(infraID string) []*ec2.Filter {
+	return []*ec2.Filter{
+		{
+			Name:   awssdk.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", infraID)),
+			Values: []*string{awssdk.String("owned")},
+		},
+		{
+			Name:   awssdk.String("tag:Name"),
+			Values: []*string{awssdk.String(awsSecurityGroupName(infraID))},
+		},
+	}
+}
+
+func awsSecurityGroupName(infraID string) string {
+	return fmt.Sprintf("%s-default-sg", infraID)
+}
+
+func createAWSDefaultSecurityGroup(ctx context.Context, ec2Client ec2iface.EC2API, infraID, vpcID string, additionalTags []hyperv1.AWSResourceTag) (string, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	// Determine VPC cidr
+	vpcResult, err := ec2Client.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		VpcIds: []*string{awssdk.String(vpcID)},
+	})
+	if err != nil {
+		logger.Error(err, "Failed to describe vpc", "vpcID", vpcID)
+		return "", fmt.Errorf("failed to describe vpc %s, code %s", vpcID, awsErrorCode(err))
+	}
+	if len(vpcResult.Vpcs) == 0 {
+		return "", fmt.Errorf("vpc %s not found", vpcID)
+	}
+	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
+	describeSGResult, err := ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{Filters: awsSecurityGroupFilters(infraID)})
+	if err != nil {
+		logger.Error(err, "Failed to list security groups")
+		return "", fmt.Errorf("cannot list security groups, code: %s", awsErrorCode(err))
+	}
+	sgID := ""
+	var sg *ec2.SecurityGroup
+	if len(describeSGResult.SecurityGroups) > 0 {
+		sg = describeSGResult.SecurityGroups[0]
+		sgID = awssdk.StringValue(sg.GroupId)
+	}
+	if sgID == "" {
+		// Create a security group if one is not found
+
+		tagKeys := sets.NewString()
+		var tags []*ec2.Tag
+		for _, tag := range additionalTags {
+			tagKeys.Insert(tag.Key)
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String(tag.Key),
+				Value: awssdk.String(tag.Value),
+			})
+		}
+		clusterKey := fmt.Sprintf("kubernetes.io/cluster/%s", infraID)
+		if !tagKeys.Has(clusterKey) {
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String(clusterKey),
+				Value: awssdk.String("owned"),
+			})
+		}
+		if !tagKeys.Has("Name") {
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String("Name"),
+				Value: awssdk.String(awsSecurityGroupName(infraID)),
+			})
+		}
+		if !tagKeys.Has("red-hat-managed") {
```

Comment:
I hear @mjlshen but fwiw this is only true for manage services. It might be confusing for consumers running in self hosted scenario I guess. I'm good keeping it for now anyways.

Reply:
hmm, you're right the default tags are going to be different in each env where hypershift is run. I will remove it. We can always add whatever tags we want via the HostedCluster API